velox/exec/tests/utils/AggregationFuzzerUtils.h

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include <boost/random/uniform_int_distribution.hpp>
+#include "velox/exec/tests/utils/InputGenerator.h"
+#include "velox/exec/tests/utils/ResultVerifier.h"
+
+namespace facebook::velox::exec::test {
+
+struct AggregationFuzzerOptions {
+  /// Comma-separated list of functions to test. By default, all functions
+  /// are tested.
+  std::string onlyFunctions;
+
+  /// Set of functions to not test.
+  std::unordered_set<std::string> skipFunctions;
+
+  /// Set of functions whose results are non-deterministic. These can be
+  /// order-dependent functions whose results depend on the order of input
+  /// rows, or functions that return complex-typed results containing
+  /// floating-point fields.
+  ///
+  /// For some functions, the result can be transformed to a deterministic
+  /// value. If such transformation exists, it can be specified to be used for
+  /// results verification. If no transformation is specified, results are not
+  /// verified.
+  ///
+  /// Keys are function names. Values are optional transformations. "{}"
+  /// should be used to indicate the original value, i.e. "f({})"
+  /// transformation applies function 'f' to aggregation result.
+  std::unordered_map<std::string, std::shared_ptr<ResultVerifier>>
+      customVerificationFunctions;
+
+  std::unordered_map<std::string, std::shared_ptr<InputGenerator>>
+      customInputGenerators;
+
+  std::unordered_set<std::string> orderDependentFunctions;
+
+  /// Timestamp precision to use when generating inputs of type TIMESTAMP.
+  VectorFuzzer::Options::TimestampPrecision timestampPrecision{
+      VectorFuzzer::Options::TimestampPrecision::kNanoSeconds};
+
+  /// A set of configuration properties to use when running query plans.
+  /// Could be used to specify timezone or enable/disable settings that
+  /// affect semantics of individual aggregate functions.
+  std::unordered_map<std::string, std::string> queryConfigs;
+};
+
+// Parse the comma separated list of function names and return a set of them.
+std::unordered_set<std::string> splitNames(const std::string& names);
+
+// Parse the comma separated list of function names, and use it to filter the
+// input signatures.
+template <typename SignatureMapType>
+SignatureMapType filterSignatures(
```

Comment:
There is similar (if not identical) function in velox/expression/tests/ExpressionFuzzer.cpp. Perhaps, move it to FuzzerToolkit.h and reuse in both places.

Reply:
@kagamiori I see that you moved filterSignatures into FuzzerToolkit.h, but I don't see changes to ExpressionFuzzer.cpp to use it. An I missing something?

Reply:
Hi @mbasmanova, the filterSignatures() function used in ExpressionFuzzer is a bit different from the one moved to FuzzerToolkit: (1) ExpressionFuzzer's filterSignature() modifies the input signature map in place rather than returning a new object, and (2) ExpressionFuzzer's filterSignature() handles function signatures in addition to function names in the skip function list. So it's not a simple replacement. Could we make this change in the separate diff since the current one is already too large?

Reply:
Sure. We can consolidate these in a follow-up. Let's add a TODO for now. Thanks.